front/pages/api/w/[wId]/labs/transcripts.ts

```diff
@@ -0,0 +1,153 @@
+
+import { Authenticator, getSession } from "@app/lib/auth";
+import type { LabsTranscriptsProviderType } from "@app/lib/labs/transcripts/utils/types";
+import { LabsTranscriptsConfigurationResource } from "@app/lib/resources/labs_transcripts_configuration_resource";
+import { apiError, withLogging } from "@app/logger/withlogging";
+import { launchRetrieveNewTranscriptsWorkflow } from "@app/temporal/labs/client";
+
+export type GetLabsTranscriptsConfigurationResponseBody = {
+  configuration: LabsTranscriptsConfigurationResource | null;
+};
+
+async function handler(
+  req: NextApiRequest,
+  res: NextApiResponse<
+    WithAPIErrorReponse<GetLabsTranscriptsConfigurationResponseBody>
+  >
+): Promise<void> {
+  const session = await getSession(req, res);
+  const auth = await Authenticator.fromSession(
+    session,
+    req.query.wId as string
+  );
+
+  const owner = auth.workspace();
+  if (!owner) {
+    return apiError(req, res, {
+      status_code: 404,
+      api_error: {
+        type: "workspace_not_found",
+        message: "The workspace was not found.",
+      },
+    });
+  }
+
+  switch (req.method) {
+    case "GET":
+      const transcriptsConfigurationGetRes =
+        await LabsTranscriptsConfigurationResource.findByUserIdAndProvider({
+          userId: owner.id,
+          provider: req.query.provider as LabsTranscriptsProviderType,
+        });
+
+      return res
+        .status(200)
+        .json({ configuration: transcriptsConfigurationGetRes });
+
+    // Update
+    case "PATCH":
+      const {
+        agentConfigurationId: patchAgentId,
+        provider: patchProvider,
+        email: emailToNotify,
+        isActive,
+      } = req.body;
+      if (!patchAgentId || !patchProvider) {
+        return apiError(req, res, {
+          status_code: 400,
+          api_error: {
+            type: "invalid_request_error",
+            message:
+              "The `connectionId` and `provider` parameters are required.",
+          },
+        });
+      }
+
+      const transcriptsConfigurationPatchRes =
+        await LabsTranscriptsConfigurationResource.findByUserIdAndProvider({
+          userId: owner.id,
+          provider: patchProvider as LabsTranscriptsProviderType,
+        });
+
+      if (!transcriptsConfigurationPatchRes) {
+        return apiError(req, res, {
+          status_code: 404,
+          api_error: {
+            type: "configuration_not_found",
+            message: "The configuration was not found.",
+          },
+        });
+      }
+
+      await LabsTranscriptsConfigurationResource.setAgentConfigurationId({
+        agentConfigurationId: patchAgentId,
+        provider: patchProvider,
+        userId: owner.id,
+      });
+
+      if (emailToNotify) {
+        await LabsTranscriptsConfigurationResource.setEmailToNotify({
+          emailToNotify,
+          provider: patchProvider,
+          userId: owner.id,
+        });
+      }
+
+      if (isActive !== undefined) {
```

Comment:
```suggestion
      if (isActive) {
```

Reply:
We need to save when isActive is set to False as well, hence the undefined check so we can't just check for isActive(). Let me know if you have a better method

Reply:
Is `isActive` required in the API's body?

Reply:
No - this is something we can discuss